### PR TITLE
Test functions and botorch utilities

### PR DESCRIPTION
**Pull request by thomaswmorris**
Not as long as the last one! And now we can test this without sirepo. I added test optimization functions and custom botorch acquisition functions.

### CODE REVIEW

**Review by mrakitin — COMMENTED** [review]
Looks good overall. I added a few suggestions below.

File: bloptools/bo/__init__.py [hunk 1]
```diff
@@ -526,36 +520,71 @@ def update_state_plots(self, gridded=False):
...
+                shading="nearest",
+            )
         else:
-            ref = ax.scatter(*self.test_params.T[:2], s=s, c=PE)
+            ref = ax.scatter(
+                *self.test_params.T[:2], c=self.classifier.entropy(self.params_trans_fun(self.test_params)), s=s
+            )
         clb = self.state_fig.colorbar(ref, ax=ax, location="bottom", aspect=32, shrink=0.8)
-        clb.set_label("nepits per volume")
+        clb.set_label("nats")
```
[the remark this answers is not in the document]
Comment (thomaswmorris) [reply bloptools/bo/__init__.py:556]: It's in information units (https://en.wikipedia.org/wiki/Nat_(unit))

File: bloptools/bo/__init__.py [hunk 2]
```diff
@@ -263,23 +268,23 @@ def recommend(
...
+            if strategy.lower() == "ei":  # greedy expected reward maximization
+                objective = acquisition.expected_improvement(evaluator, classifier, TEST_X).sum(axis=1)
 
             if strategy.lower() == "explore":  # greedy expected reward maximization
                 objective = -self._negative_expected_variance(evaluator, classifier, TEST_X).sum(axis=1)
 
-            # if strategy.lower() == "eig":
-            #     objective = -_negative_expected_information_gain(evaluator, classifier, evaluator.X, TEST_X)
+            if strategy.lower() == "egibbon":
+                objective = acquisition.expected_gibbon(evaluator, classifier, TEST_X).sum(axis=1)
```
[the remark this answers is not in the document]
Comment (thomaswmorris) [reply bloptools/bo/__init__.py:278]: https://arxiv.org/abs/2102.03324

[the remark this answers is not in the document]
Comment (thomaswmorris) [reply bloptools/bo/__init__.py:278]: It's a fancy acquisition metric that's already nicely implemented in botorch

**Review by AbbyGi — APPROVED** [review]

**Review by mrakitin — APPROVED** [review]
I have a suggestion about a dynamically-created ophyd `Device`, which can be implemented in the next PR. This one looks good!

**Pull request merged**